### PR TITLE
Get sign of an instruction

### PR DESCRIPTION
**Pull request by thanhtoantnt**
# Description
C binding to get a sign of an instruction

In process of fixing https://github.com/sbip-sg/verazt/issues/85

# How Has This PR Been Tested?

Please describe the tests that you ran to verify your changes. Please also note
any relevant details for your test configuration.

- The binding is tested with the input files from benchmarks of [discover](https://github.com/sbip-sg/verazt )

# Checklist:

Please ensure that all items are reviewed and checked by the notation: `[x]`.
(The irrelevant items can be stricken through by the notation: `~~ ... ~~`, but
they still need to be checked.)

- [x] Self-review: I have reviewed my code thoroughly before creating the PR.
- [x] Code linting: I have run `clang-format` with relevant files.

### CODE REVIEW

File: llvm/lib/IR/Core.cpp [hunk 1]
```diff
@@ -2893,6 +2894,34 @@ LLVMBool LLVMHasNoSignedWrap(LLVMValueRef Inst) {
...
 }
 
+int LLVMGetSignInfo(LLVMValueRef Inst){
+  DbgDeclareInst* dbg_declare_inst = dyn_cast<DbgDeclareInst>(unwrap(Inst));
+  if (!dbg_declare_inst) {
+    return -1;
+  }
+
+  DILocalVariable* local_var = dbg_declare_inst->getVariable();
+
```
Comment (taquangtrung) [inline llvm/lib/IR/Core.cpp]: Remove this  empty line?

File: llvm/lib/IR/Core.cpp [hunk 2]
```diff
@@ -2893,6 +2894,34 @@ LLVMBool LLVMHasNoSignedWrap(LLVMValueRef Inst) {
...
+  DIType* type_info = local_var->getType();
+
```
Comment (taquangtrung) [inline llvm/lib/IR/Core.cpp]: Remove this empty line?

File: llvm/lib/IR/Core.cpp [hunk 3]
```diff
@@ -2893,6 +2894,34 @@ LLVMBool LLVMHasNoSignedWrap(LLVMValueRef Inst) {
   return false;
 }
 
+int LLVMGetSignInfo(LLVMValueRef Inst){
+  DbgDeclareInst* dbg_declare_inst = dyn_cast<DbgDeclareInst>(unwrap(Inst));
```
Comment (taquangtrung) [inline llvm/lib/IR/Core.cpp]: The naming style of variable names in this function isn't consistent with other functions in this file.
@thanhtoantnt: can you refactor the code?

**Review by taquangtrung — CHANGES_REQUESTED** [review]

File: llvm/lib/IR/Core.cpp [hunk 4]
```diff
@@ -2893,6 +2894,34 @@ LLVMBool LLVMHasNoSignedWrap(LLVMValueRef Inst) {
   return false;
 }
 
+int LLVMGetSignInfo(LLVMValueRef Inst){
```
Comment (taquangtrung) [inline llvm/lib/IR/Core.cpp]: Rename function to `LLVMGetSignednessInfo`

**Review by taquangtrung — CHANGES_REQUESTED** [review]

**Comment by thanhtoantnt** [comment]
@taquangtrung I refactored the code to resolve your comments.

File: llvm/include/llvm-c/Core.h [hunk 5]
```diff
@@ -1561,95 +1558,47 @@ LLVMTypeRef LLVMX86AMXType(void);
...
+          ConstantFP) macro(ConstantInt) macro(ConstantPointerNull) macro(ConstantStruct)           \
+          macro(ConstantTokenNone) macro(ConstantVector) macro(GlobalValue) macro(                  \
+              GlobalAlias) macro(GlobalObject) macro(Function) macro(GlobalVariable)                \
+              macro(GlobalIFunc) macro(UndefValue) macro(PoisonValue) macro(Instruction) macro(     \
+                  UnaryOperator) macro(BinaryOperator) macro(CallInst) macro(IntrinsicInst)         \
+                  macro(DbgInfoIntrinsic) macro(DbgVariableIntrinsic) macro(                        \
+                      DbgDeclareInst) macro(DbgLabelInst) macro(MemIntrinsic)                       \
+                      macro(MemCpyInst) macro(MemMoveInst) macro(MemSetInst) macro(                 \
+                          CmpInst) macro(FCmpInst) macro(ICmpInst) macro(ExtractElementInst)        \
+                          macro(GetElementPtrInst) macro(InsertElementInst) macro(                  \
```
Comment (taquangtrung) [inline llvm/include/llvm-c/Core.h]: `Clangformat` seems to break all default formatting of the old code base?

@thanhtoantnt: can you investigate how to run Clangformat but still preserve the existing code structure?

If this is not possible, then we may need to formatting the code manually.

**Review by taquangtrung — CHANGES_REQUESTED** [review]

File: llvm/include/llvm-c/Core.h (hunk 5 above)
Comment (taquangtrung) [reply llvm/include/llvm-c/Core.h]: @thanhtoantnt: maybe this can help: https://clang.llvm.org/docs/ClangFormatStyleOptions.html

**Comment by thanhtoantnt** [comment]
@taquangtrung Now, only new code is formatted using `clang-format`, other parts are kept intact.

**Review by taquangtrung — APPROVED** [review]

**Comment by taquangtrung** [comment]
Merged! Thanks @thanhtoantnt!

**Pull request merged**